zero_bin/common/src/block_interval.rs

```diff
@@ -108,6 +108,20 @@ impl BlockInterval {
         }
     }
 
+    pub fn get_start_block(&self) -> Result<u64> {
+        match self {
+            BlockInterval::SingleBlockId(BlockId::Number(num)) => {
+                let num_value = num
+                    .as_number()
+                    .ok_or_else(|| anyhow!("invalid block number '{num}'"))?;
+                Ok(num_value) // Return the valid block number
+            }
+            BlockInterval::Range(range) => Ok(range.start),
+            BlockInterval::FollowFrom { start_block, .. } => Ok(*start_block),
+            _ => Err(anyhow!("Unknown BlockInterval variant")), // Handle unknown variants
```

Comment:
You don't need this one. If in theory we add new variant for `BlockInterval`, compiler would warn us to add new case for this match

Reply:
actually it is complaining right now `BlockInterval::SingleBlockId()` can contain a hash as well instead of num and currently we dont support it so

Reply:
maybe we can handle case of hash differently